### PR TITLE
Add access key to context menus

### LINKED ISSUE #88: Access keys in context menus
https://discourse.mozilla.org/t/how-to-create-a-menu-with-underscore-as-key-binding/28730

### PR DESCRIPTION
**Pull request by linusaurusrex**
Fixes #88 
Add the underscore as an access key in Firefox 63+.

### CODE REVIEW

**Comment by rugk** [comment]
Hi @linusaurusrex,
thanks for your first contribution to this project! :tada: :+1: 
I hope you'll like this project and enjoy hacking on it… :smiley:

**Comment by rugk** [comment]
@linusaurusrex 
BTW I want to merge and release this with v1.3 in October still.

**Comment by linusaurusrex** [comment]
1. I used `_` because the link you posted in #88 mentioned an underscore as an access key, but `Q` makes more sense. 
2. Why do you suggest adding a "ContextMenu" module when one already exists?

File: src/popup/modules/UserInterface.js [hunk 1]
```diff
@@ -558,9 +558,9 @@ export function init() {
         }
 
         // create save menu if needed
-        browser.menus.create({
+        browser.runtime.getBrowserInfo().then(({version}) => browser.menus.create({
```
[the remark this answers is not in the document]
Comment (rugk) [reply src/popup/modules/UserInterface.js]: This is still open…

**Comment by rugk** [comment]
Yeah, the ContextMenu module is only for the `background` part. As you also use it in the popup dir, however, it needs to be used by both, so better put it in the `commons` dir.

File: src/common/modules/ContextMenu.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import * as Logger from "./Logger.js";
+
+const browserInfo = browser.runtime.getBrowserInfo();
+
+/**
+ * Creates a context menu item
+ * 
+ * @function
```
Comment (rugk) [inline src/common/modules/ContextMenu.js]: I know some old JSDOC still document it like this in this project here, but I'd rather note that this is public, i.e.:
```suggestion
 * @public
```

File: src/popup/modules/UserInterface.js [hunk 3]
```diff
@@ -554,13 +555,12 @@ export function init() {
             // remove menu item if it has been added before
             browser.menus.remove(CONTEXT_MENU_SAVE_IMAGE);
 
-            return;
+            return undefined;
```
Comment (rugk) [inline src/popup/modules/UserInterface.js]: This says not much and you could also have returned nothing AFAIK. However, what about specifically saying that the promise is resolved here:
```suggestion
            return Promise.resolve();
```

**Pull request merged**